src/builder/create_embed.rs

```diff
@@ -316,77 +286,29 @@ impl Default for CreateEmbed {
             description: None,
             thumbnail: None,
             timestamp: None,
-            kind: "rich",
+            kind: Some("rich".to_string()),
             author: None,
             colour: None,
             footer: None,
             image: None,
             title: None,
             url: None,
+            provider: None,
+            video: None,
         }
     }
 }
 
-impl From<Embed> for CreateEmbed {
-    /// Converts the fields of an embed into the values for a new embed builder.
-    ///
-    /// Some values - such as Proxy URLs - are not preserved.
-    fn from(embed: Embed) -> Self {
-        let mut b = CreateEmbed {
-            description: embed.description,
-            timestamp: embed.timestamp,
-            title: embed.title,
-            url: embed.url,
-            ..Default::default()
-        };
-
-        if let Some(colour) = embed.colour {
-            b = b.colour(colour);
-        }
-
-        if let Some(thumbnail) = embed.thumbnail {
-            b = b.thumbnail(thumbnail.url);
-        }
-
-        if let Some(image) = embed.image {
-            b = b.image(image.url);
-        }
-
-        if let Some(author) = embed.author {
-            b = b.author(author.into());
-        }
-
-        if let Some(footer) = embed.footer {
-            b = b.footer(footer.into());
-        }
-
-        for field in embed.fields {
-            b = b.field(field.name, field.value, field.inline);
-        }
-
-        b
-    }
-}
-
-/// A builder to create a fake [`Embed`] object's author, for use with the [`CreateEmbed::author`]
-/// method.
-#[derive(Clone, Debug, Serialize)]
-#[must_use]
-pub struct CreateEmbedAuthor {
-    name: String,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    icon_url: Option<String>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    url: Option<String>,
-}
+pub type CreateEmbedAuthor = EmbedAuthor;
 
-impl CreateEmbedAuthor {
+impl EmbedAuthor {
```

Comment:
```suggestion
impl CreateEmbedAuthor {
```
To be consistent with the rest of the builders, which impl on the type alias instead of the original type.